react-web/src/pages/subscription/history/SubscriptionHistory.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from "react";
+import { useAppSelector } from "store/store";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import tz from "dayjs/plugin/timezone";
+
+import TableComponent from "components/Table/Table";
+import { fetchData } from "api/api";
+import { Subscription } from "../Subscription.interface";
+import Toast from "components/Toast/Toast";
+import { getCurrentAdaUsdPrice } from "store/slices/auth.slice";
+
+dayjs.extend(utc)
+dayjs.extend(tz)
+
+const SubscriptionHistory = () => {
+    const dispatch = useDispatch()
+    const navigate = useNavigate();
+    const {adaUsdPrice} = useAppSelector((state) => state.auth);
+    const [data, setData] = useState<Array<Subscription>>([]);
+    const [skipPageReset, setSkipPageReset] = useState(false);
+    const [errorToast, setErrorToast] = useState<{display: boolean; statusText?: string; message?: string;}>({display: false});
+    const timeZone = dayjs.tz.guess()
+
+    useEffect(() => {
+        fetchTableData();
+        if (adaUsdPrice === 0) {
+            dispatch(getCurrentAdaUsdPrice())
+        }
+    }, []);
+    
+    useEffect(() => {
+        setSkipPageReset(false);
+    }, [data]);
+
+    const handleError = (error: any) => {
+        if (error.response) {
+          setErrorToast({display: true, statusText: error.response.statusText, message: error.response.data || undefined})
+        } else {
+          setErrorToast({display: true})
+        }
+        const timeout = setTimeout(() => { clearTimeout(timeout); setErrorToast({display: false}) }, 3000)
+    }
+  
+      
+    const updateMyData = (rowIndex: any, columnID: any, value: any) => {
+        setSkipPageReset(true); // turn on flag to not reset the page
+        setData((old) =>
+          old.map((row, index) => {
+            if (index === rowIndex) {
+              return {
+                ...old[rowIndex],
+                [columnID]: value,
+              };
+            }
+            return row;
+          })
+        );
```

Comment:
this creates some conflict in the reasoning:
- at line 49 we have 'setSkipPageReset(true)`
- at line 50 we change the `data` state, and because of the `useEffect` from line 34 we have `setSkipPageReset(false)`

which is it: true or false?

Reply:
skipPageReset should be false, but for a meanwhile we have to make it true until we set the data to the cells.

Reply:
> at line 50 we change the data state, and because of the useEffect from line 34 we have setSkipPageReset(false)
what I mean is that is a confusing behavior, because we have 2 setSkipPageReset calls: 
1) 49 setting it to  true
2) another one to false, at line 35 triggered by line 50

Reply:
Seems like this bit was unnecessary here. Removed the code